### PR TITLE
support for error messages

### LINKED ISSUE #24: include error messages in output
I set error messages with the `withMessage` `io-ts-types` helper as a guide, but PathReporter doesn't output them.

### PR DESCRIPTION
**Pull request by osdiab**
Closes #24 , also removes the trailing period for the provided value (i think it doesn't help/the period can hurt clarity).

~~built on top of #20 so there's noise in the diff, here are the changes without that: https://github.com/OliverJAsh/io-ts-reporters/pull/25/files/39716c36ea5205155c0ba7117a5218940d90d983..57f0b2ae54130cd428cfd634978add1bfe983e9d~~

I rebased to make it based on master.

Example output at https://github.com/OliverJAsh/io-ts-reporters/pull/25/files#diff-e1b1ef74bf2611df9cd1e2173b684deeR85

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -20,16 +20,21 @@ export const formatValidationError = (error: t.ValidationError) => {
...
+        maybeErrorContext,
+        map((errorContext) => {
+            const expectedType = errorContext.type.name;
+            return (
+                // https://github.com/elm-lang/core/blob/18c9e84e975ed22649888bfad15d1efdb0128ab2/src/Native/Json.js#L199
+                // tslint:disable-next-line:prefer-template
+                `Expecting ${expectedType}` +
+                (path === '' ? '' : ` at ${path}`) +
+                ` but instead got: ${jsToString(error.value)}${
+                error.message ? ` (message: ${jsToString(error.message)})` : ''
```
Comment (gillchristian) [inline src/index.ts]: I'm not so sure about the `(message: ....)` format, what about just having the message in parens? Wdyt?

```suggestion
                error.message ? ` (${jsToString(error.message)})` : ''
```

Also, the formatting here inside template literal is getting a bit crazy. Maybe this is worth extracting.

Comment (osdiab) [reply src/index.ts]: I'm OK either way! I'll remove it and look into simplifying the template.

Comment (osdiab) [reply src/index.ts]: personally though I don't really mind the template string haha

**Comment by osdiab** [comment]
refactored, ready to go

**Comment by gillchristian** [comment]
@osdiab this needs merge conflicts resolving as well

**Comment by osdiab** [comment]
it doesn't show any merge conflicts with master? @gillchristian

**Comment by gillchristian** [comment]
There were conflicts for rebasing and merging, but squash and merge worked :)

**Pull request merged**